dvc/remote/base.py

```diff
@@ -677,18 +677,31 @@ def safe_remove(self, path_info, force=False):
     def _checkout_file(
         self, path_info, checksum, force, progress_callback=None
     ):
-        cache_info = self.checksum_to_path_info(checksum)
-        if self.exists(path_info):
-            msg = "data '{}' exists. Removing before checkout."
-            logger.warning(msg.format(str(path_info)))
-            self.safe_remove(path_info, force=force)
+        if self._needs_checkout(path_info, checksum):
+            if self.exists(path_info):
+                msg = "Data '{}' exists. Removing before checkout."
+                logger.warning(msg.format(str(path_info)))
+                self.safe_remove(path_info, force=force)
```

Comment:
Also, do we need this message? Because if we don't then we can save `.exists()` call here and simply call `.safe_remove()` unconditionally.

Reply:
Well the question is how does it look like from ui POV. I think its not necessary, since `safe_remove` passes the information that file already exists.